ROADMAP.md

```diff
@@ -1,64 +1,106 @@
+#### End of Q4 (December)
+-   B.1 🧠 Smart Dialing - **Happy Eyeballs**
+-   C.1 📊 Comprehensive Metrics - **Swarm metrics**
+-   D.1 📺 Universal Browser Connectivity - **[WebRTC](https://github.com/libp2p/go-libp2p/pull/1655)**
 
-How: This will look similar to how we already expose resource manager metrics. Metrics can be added incrementally for libp2p’s components. First milestone is having metrics for the swarm.
+### 2023
 
-Tracking issue: [https://github.com/libp2p/go-libp2p/issues/1356](https://github.com/libp2p/go-libp2p/issues/1356)
+#### Early Q1 (January)
+-   B.2 🧠 Smart Dialing - **Quic Blackhole detector**
+-   E 🧪 Future-proof testing
+-   A.2 ⚡ Handshakes at the Speed of Light - **[Adding security protocol](https://github.com/libp2p/specs/pull/353)**
 
-## 📺 Universal Browser Connectivity
+#### Mid Q1 (February)
+-   B.3 🧠 Smart Dialing - **RTT estimation**
+    -   🎉 Estimated Project Completion
 
-Why: A huge part of “the Web” is happening inside the browser. As a universal p2p networking stack, libp2p needs to be able to offer solutions for browser users.
+#### End of Q1 (March)
+-   F ⏱️ Future-proof benchmarking
+-   A.3 ⚡ Handshakes at the Speed of Light - **.5 RTT data optimization**
+    -   🎉 Estimated Project Completion
 
-Goal: go-libp2p ships with up-to-date WebTransport and (libp2p-) WebRTC implementations, enabled by default. This allows connections between browsers and public nodes, browsers and non-public nodes as well as two browsers.
+#### Sometime Q2
+-   G 📢 Judicious Address Advertisements
 
-1. WebTransport: as the protocol is still under development by IETF and W3C, our implementation needs to follow. To stay up to date, we will have to move as soon as Chrome ships support for a new draft version
-2. WebRTC: while browser to public node is getting close to finalized, there’ll be a push to make the other combinations work as well
+#### Yet to be planned/TBD
+-   D.2 📺 Universal Browser Connectivity - **WebTransport**
+-   H 🧗 Clearing our tech debt
 
-## ⚡️Handshakes at the Speed of Light
+# Roadmap Appendix
+## A.⚡️Handshakes at the Speed of Light
+<!--- TODO: Link to GitHub Epic -->
 
-Why: Historically, libp2p has been very wasteful when it comes to round trips spent during connection establishment. This is slowing down our users, especially their TTFB (time to first byte) metrics.
+**Why**: Historically, libp2p has been very wasteful when it comes to round trips spent during connection establishment. This is slowing down our users, especially their TTFB (time to first byte) metrics.
 
-Goal: go-libp2p optimizes its handshake latency up to the point where only increasing the speed of light would lead to further speedups. In particular, this means (in chronological order of a handshake):
+**Goal**: go-libp2p optimizes its handshake latency up to the point where only increasing the speed of light would lead to further speedups. In particular, this means (in chronological order of a handshake):
```

Comment:
```suggestion
**Goal**: go-libp2p optimizes its handshake latency up to the point where only increasing the speed of light would lead to further speedups. In particular, this means:
```